src/rdkafka_admin.c

```diff
@@ -7734,3 +7848,569 @@ rd_kafka_DescribeConsumerGroups_result_groups(
+    size_t idx) {
+        rd_kafka_AclOperation_t *entry =
+            rd_list_elem(&topicdesc->authorized_operations, idx);
+        return *entry;
+}
+
+const int rd_kafka_TopicDescription_topic_authorized_operation_count(
+    const rd_kafka_TopicDescription_t *topicdesc) {
+        return rd_list_cnt(&topicdesc->authorized_operations);
+}
+
+const int rd_kafka_TopicDescription_partition_id(
+    const rd_kafka_TopicDescription_t *topicdesc,
+    int idx) {
+        return topicdesc->partitions[idx].id;
+}
+
+const int rd_kafka_TopicDescription_partition_leader(
+    const rd_kafka_TopicDescription_t *topicdesc,
+    int idx) {
+        return topicdesc->partitions[idx].leader;
+}
+
+const int rd_kafka_TopicDescription_partition_isr_count(
+    const rd_kafka_TopicDescription_t *topicdesc,
+    int idx) {
+        return topicdesc->partitions[idx].isr_cnt;
+}
+
+const int rd_kafka_TopicDescription_partition_replica_count(
+    const rd_kafka_TopicDescription_t *topicdesc,
+    int idx) {
+        return topicdesc->partitions[idx].replica_cnt;
+}
+
+const int rd_kafka_TopicDescription_partition_isr(
+    const rd_kafka_TopicDescription_t *topicdesc,
+    int partition_idx,
+    int isr_idx) {
+        return topicdesc->partitions[partition_idx].isrs[isr_idx];
+}
+
+const int rd_kafka_TopicDescription_partition_replica(
+    const rd_kafka_TopicDescription_t *topicdesc,
+    int partition_idx,
+    int replica_idx) {
+        return topicdesc->partitions[partition_idx].replicas[replica_idx];
+}
+
+const rd_kafka_resp_err_t rd_kafka_TopicDescription_partition_error(
+    const rd_kafka_TopicDescription_t *topicdesc,
+    int idx) {
+        return topicdesc->partitions[idx].err;
+}
+
+const int rd_kafka_TopicDescription_topic_partition_count(
+    const rd_kafka_TopicDescription_t *topicdesc) {
+        return topicdesc->partition_cnt;
+}
+
+const char *rd_kafka_TopicDescription_topic_name(
+    const rd_kafka_TopicDescription_t *topicdesc) {
+        return topicdesc->topic;
+}
+
+const rd_kafka_error_t *
+rd_kafka_TopicDescription_error(const rd_kafka_TopicDescription_t *topicdesc) {
+        return topicdesc->error;
+}
+
+const rd_kafka_TopicDescription_t **rd_kafka_DescribeTopics_result_topics(
+    const rd_kafka_DescribeTopics_result_t *result,
+    size_t *cntp) {
+        const rd_kafka_op_t *rko = (const rd_kafka_op_t *)result;
+        rd_kafka_op_type_t reqtype =
+            rko->rko_u.admin_result.reqtype & ~RD_KAFKA_OP_FLAGMASK;
+        rd_assert(reqtype == RD_KAFKA_OP_DESCRIBETOPICS);
+
+        *cntp = rd_list_cnt(&rko->rko_u.admin_result.results);
+        return (const rd_kafka_TopicDescription_t **)
+            rko->rko_u.admin_result.results.rl_elems;
+}
+
+/**
+ * @brief Helper function to clear a rd_kafka_metadata_partition.
+ *
+ * @note Does not deallocate the rd_kafka_metadata_partition itself.
+ */
+static void
+rd_kafka_metadata_partition_clear(struct rd_kafka_metadata_partition *rkmp) {
```

Comment:
There are no _new or _destroys for this struct (since we almost always use it inside a pre-allocated tmpabuf).
I'm moving this function and the copy function to the metadata.c file where it might make most sense.

Reply:
Will check for other new types and destroy/new functions